client/certbot-dns-ipa.in

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+"""
+This script can be used with Certbot to satisfy dns-01 challenges.
+FreeIPA integrated DNS is required.
+
+This script can be used for both --manual-auth-hook and
+--manual-cleanup-hook.
+
+"""
+
+import os
+from dns import resolver
+from ipalib import api
+from ipapython import dnsutil
+
+certbot_domain = os.environ['CERTBOT_DOMAIN']
+certbot_validation = os.environ['CERTBOT_VALIDATION']
+if 'CERTBOT_AUTH_OUTPUT' in os.environ:
+    command = 'dnsrecord_del'
+else:
+    command = 'dnsrecord_add'
+
+validation_domain = f'_acme-challenge.{certbot_domain}'
+fqdn = dnsutil.DNSName(validation_domain).make_absolute()
+zone = dnsutil.DNSName(resolver.zone_for_name(fqdn))
+name = fqdn.relativize(zone)
+
+api.bootstrap(context='cli')
+api.finalize()
+api.Backend.rpcclient.connect()
+
+api.Command[command](zone, name, txtrecord=[certbot_validation], dnsttl=60)
```

Comment:
It should check whether it has Kerberos credentials and fail in some graceful way if it doesn't. It currently fails with several very long and scary tracebacks.

It should be documented somewhere what role and/or permissions are required for this to work.

Reply:
Tracebacks are gone, added some more doc to the script about what is required and its intended use (i.e. intended for use only as a Certbot hook script).  Updates in next push!